src/components/CurrencyInput.tsx

```diff
@@ -1,36 +1,62 @@
-import React from 'react';
-import PropTypes from 'prop-types';
+import React, { FunctionComponent } from 'react';
 import classnames from 'classnames';
 import MaskedInput from 'react-text-mask';
 import createNumberMask from 'text-mask-addons/dist/createNumberMask';
 import InputGroup from './InputGroup';
 import InputGroupAddon from './InputGroupAddon';
 
+type Config = {
+  guide?: boolean,
+  previousConformedValue: string,
+  placeholderChar?: string,
+  placeholder?: string,
+  currentCaretPosition?: number,
+  keepCharPositions?: boolean,
+  rawValue: string,
+}
+
 /**
  * In the case where the user enters an extra "." at the end of the input the default behavior will
  * be to remove the original decimal point and keep the new one, resulting in the value being
  * multiplied by 100.  If we detect an additional decimal point we can ignore the extra character.
  */
-function preventMultipleDecimalPoint(conformedValue, config) {
+function preventMultipleDecimalPoint(conformedValue: string, config: Config): string {
   let result = conformedValue;
   if (config.rawValue.match(/\..*\./)) {
     result = config.previousConformedValue;
   }
   return result;
 }
 
+type Props = {
+  allowDecimal?: boolean;
+  allowNegative?: boolean;
+  className?: string;
+  includeThousandsSeparator?: boolean;
+  inputProps?: { className?: string };
+  size?: string;
+  state?: any;
+  type?: any;
+};
```

Comment:
The original type declaration for these props included `React.InputHTMLAttributes<HTMLInputElement>`:
https://github.com/appfolio/react-gears/blob/039ca6a4f058e91cd0e3dd2a40d170224c879120/src/components/CurrencyInput.d.ts#L4
which allowed props like `value`, `onChange`, `onBlur`, etc. to be passed, but this new type definition doesn't allow that.

Reply:
This is fixed in https://github.com/appfolio/react-gears/pull/814